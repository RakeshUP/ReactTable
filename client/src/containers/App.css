.App {
  text-align: center;
}

table {
  margin: 2vh auto;
}

td {
  border: 1px solid #333;
}

label {
  padding-left: 2vw;
}

.filter {
  padding-top: 2vh;
}