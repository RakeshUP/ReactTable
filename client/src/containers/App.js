import React, { useState } from 'react';
import uploadFile from '../util/uploadFile';

import './App.css';

const RenderRows = ({ row, delimiter, lineNumber }) => (
  <tr key={lineNumber}>
    {row.split(delimiter).map((column, index) => <td key={index}>{column}</td>)}
  </tr>
);

const App = (props) => {
  const initialFileData = props.fileData || '';
  const [fileData, setFileData] = useState(initialFileData);
  const [delimiter, setDelimiter] = useState('|');
  const [lines, setLines] = useState('2');

  const fileRows = fileData.split('\n');

  const handleSubmit = async (event) => {
    event.preventDefault();
    const formData = new FormData();

    const fileField = document.querySelector('input[type="file"]');
    formData.append('file', fileField.files[0]);

    uploadFile(formData).then(setFileData);
  }

  const handleDrop = event => {
    event.stopPropagation();
    event.preventDefault();

    const formData = new FormData();
    formData.append('file', event.dataTransfer.files[0]);

    uploadFile(formData).then(setFileData);
  }

  const preventDefaultAction = e => {
    e.stopPropagation();
    e.preventDefault();
  }

  return (
    <div className="App" onDragEnter={preventDefaultAction}
      onDragOver={preventDefaultAction} onDrop={handleDrop}>
      <form className="form" onSubmit={handleSubmit}>
        <div>
          <label htmlFor="file">Choose file to upload: </label>
          <input type="file" id="file" name="file" accept=".txt" />
        </div>
        <div>
          <button id="submit">Submit</button>
        </div>
      </form>
      <div className="filter">
        <label htmlFor="delimiter">Delimiter:</label>
        <input type="text" id="delimiter" name="delimiter" defaultValue={delimiter}
          onChange={(event) => setDelimiter(event.target.value)} />
        <label htmlFor="lines">Lines:</label>
        <input type="text" id="lines" name="lines" defaultValue={lines}
          onChange={(event) => setLines(event.target.value)} />
      </div>
      <table>
        <tbody>
          {fileRows.map((row, lineNumber) => lineNumber < lines
            && <RenderRows key={lineNumber} row={row} delimiter={delimiter} lineNumber={lineNumber} />
          )}
        </tbody>
      </table>
    </div >
  );
}

export default App;
