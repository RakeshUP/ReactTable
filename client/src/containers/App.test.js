import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

const sampleData = `Jim Smith|123 Any Street|Boston|US|02134
Jane Lee | 248 Another St.| Boston | US | 02130`;

test('renders the table according to snapshot', () => {
  const { container } = render(<App fileData={sampleData} />);
  expect(container.firstChild).toMatchSnapshot();
});
