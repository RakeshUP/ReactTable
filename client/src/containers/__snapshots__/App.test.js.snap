// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the table according to snapshot 1`] = `
<div
  class="App"
>
  <form
    class="form"
  >
    <div>
      <label
        for="file"
      >
        Choose file to upload: 
      </label>
      <input
        accept=".txt"
        id="file"
        name="file"
        type="file"
      />
    </div>
    <div>
      <button
        id="submit"
      >
        Submit
      </button>
    </div>
  </form>
  <div
    class="filter"
  >
    <label
      for="delimiter"
    >
      Delimiter:
    </label>
    <input
      id="delimiter"
      name="delimiter"
      type="text"
      value="|"
    />
    <label
      for="lines"
    >
      Lines:
    </label>
    <input
      id="lines"
      name="lines"
      type="text"
      value="2"
    />
  </div>
  <table>
    <tbody>
      <tr>
        <td>
          Jim Smith
        </td>
        <td>
          123 Any Street
        </td>
        <td>
          Boston
        </td>
        <td>
          US
        </td>
        <td>
          02134
        </td>
      </tr>
      <tr>
        <td>
          Jane Lee 
        </td>
        <td>
           248 Another St.
        </td>
        <td>
           Boston 
        </td>
        <td>
           US 
        </td>
        <td>
           02130
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
